### PR TITLE
Preserve client address down to Request.t

### PR DESCRIPTION
**Pull request by vphantom**
It was as difficult as predicted, and I'm pretty sure you won't like it as-is, but this does preserve the client address, when available, down to `Request.t`.

### CODE REVIEW

**Comment by c-cube** [comment]
Ok so, why is the address optional in `Request.t` ? :)

File: src/Tiny_httpd_server.mli [hunk 1]
```diff
@@ -67,6 +67,7 @@ module Request : sig
     meth: Meth.t;  (** HTTP method for this request. *)
     host: string;
         (** Host header, mandatory. It can also be found in {!headers}. *)
+    client_addr : Unix.sockaddr option;  (** Client address. *)
```
Comment (c-cube) [inline src/Tiny_httpd_server.mli:70]:
```suggestion
    client_addr : Unix.sockaddr option;  (** Client address. Available since NEXT_RELEASE. *)
```

**Comment by vphantom** [comment]
> Ok so, why is the address optional in `Request.t` ? :)

Because `Internal_` does not use an address.

**Comment by vphantom** [comment]
I haven't found anyone actually using it so I didn't dare touch it. Is it related to that `(*$R` comment? Is that comment actually test code?

If so, then do I hard-code localhost in that comment or in `Internal_` itself? Author's choice. 😉

**Comment by vphantom** [comment]
Option gone, localhost hard-coded in the test case. How's it look now?

**Comment by c-cube** [comment]
merged manually.

**Pull request closed without merging**